lib/engine/game/g_18_uruguay/step/route.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/route'
+
+module Engine
+  module Game
+    module G18Uruguay
+      module Step
+        class Route < Engine::Step::Route
+          def setup
+            @goods_train = nil
+            @round.current_routes = []
+
+            @train_goods_combos = []
+          end
+
+          def actions(entity)
+            return %w[run_routes choose].freeze if entity.corporation == @game.rptla
+            return [] if !entity.operator? || @game.route_trains(entity).empty? || !@game.can_run_route?(entity)
+            return [] if entity.corporation? && entity.type == :minor
+
+            actions = ACTIONS.dup
+            actions << 'choose' if choosing?(entity)
+            actions
+          end
+
+          def route_for_node(entity)
+            @routes = []
+            route = Engine::Route.new(
+              @game,
+              @game.phase,
+              entity.trains.first,
+              revenue: 999,
+              revenue_str: 'aaa',
+              routes: @routes
+            )
+            route.touch_node(@game.hex_by_id('A9').tile.cities.first)
+            route.touch_node(@game.hex_by_id('A11').tile.offboards.first)
+
+            route
+          end
+
+          def choosing?(_entity)
+            true
+          end
+
+          def attach_good_to_train(train, hex)
+            train.name += '+' + @game.class::GOODS_TRAIN + '(' + hex.id + ')' if hex
```

Comment:
A lot of the code relating to train names is hard to read and understand, especially the regex parsing stuff. Is there another way that data about goods on the train could be stored, like an instance variable on the game class?